pkg/storage/snapshot/snapshot_test.go

```diff
@@ -1924,6 +1934,102 @@ var _ = Describe("Snapshot controlleer", func() {
 
 				Expect(updateCalled).To(BeTrue())
 			})
+			Context("with an instancetype or preference referenced by the VirtualMachine", func() {
+
+				var (
+					vm                           *v1.VirtualMachine
+					vmSnapshot                   *snapshotv1.VirtualMachineSnapshot
+					instancetypeObj              *instancetypev1alpha2.VirtualMachineInstancetype
+					preferenceObj                *instancetypev1alpha2.VirtualMachinePreference
+					instancetypeCR, preferenceCR *appsv1.ControllerRevision
+					err                          error
+				)
+
+				BeforeEach(func() {
+
+					vmSnapshot = createVMSnapshotInProgress()
+					vm = createLockedVM()
+
+					expectedSnapshotUpdate := vmSnapshot.DeepCopy()
+					expectedSnapshotUpdate.ResourceVersion = "1"
+					expectedSnapshotUpdate.Status = &snapshotv1.VirtualMachineSnapshotStatus{
+						SourceUID:  &vmUID,
+						ReadyToUse: &f,
+						Phase:      snapshotv1.InProgress,
+						Conditions: []snapshotv1.Condition{
+							newProgressingCondition(corev1.ConditionTrue, "Source locked and operation in progress"),
+							newReadyCondition(corev1.ConditionFalse, "Not ready"),
+						},
+						Indications: []snapshotv1.Indication{},
+					}
+
+					expectedSnapshotUpdate2 := expectedSnapshotUpdate.DeepCopy()
+					expectedSnapshotUpdate.ResourceVersion = "2"
+
+					expectVMSnapshotUpdate(vmSnapshotClient, expectedSnapshotUpdate2)
+					expectVMSnapshotUpdate(vmSnapshotClient, expectedSnapshotUpdate)
+
+					instancetypeObj = createInstancetype()
+					instancetypeCR, err = instancetype.CreateControllerRevision(vm, instancetypeObj)
+					Expect(err).ToNot(HaveOccurred())
+					crSource.Add(instancetypeCR)
+
+					preferenceObj = createPreference()
+					preferenceCR, err = instancetype.CreateControllerRevision(vm, preferenceObj)
+					Expect(err).ToNot(HaveOccurred())
+					crSource.Add(preferenceCR)
+
+				})
+
+				It("should capture ControllerRevision clone of instancetype and reference it from VirtualMachineSnapshotContent", func() {
```

Comment:
The tests look very similar, can you make them a table?

Reply:
ACK done.